core/src/main/java/org/arquillian/smart/testing/impl/StrategyDependencyResolver.java

```diff
@@ -0,0 +1,68 @@
+package org.arquillian.smart.testing.impl;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+public class StrategyDependencyResolver {
+
+    protected static final String SMART_TESTING_STRATEGY_PREFIX = "smart.testing.strategy.";
+
+    private final Path propertiesPath; // TODO this could be configurable through system property and with this we need a path
+
+    public StrategyDependencyResolver(Path propertiesPath) {
+        this.propertiesPath = propertiesPath;
+    }
+
+    public StrategyDependencyResolver() {
+        this.propertiesPath = null;
+    }
+
+    public List<String> resolveStrategies() {
+        final Properties properties = new Properties();
+        properties.putAll(loadDefaultMapping());
+        properties.putAll(loadFromFile());
+        properties.putAll(System.getProperties());
+        return transformToStrategies(properties);
+    }
+
+    protected List<String> transformToStrategies(Properties properties) {
+        return properties
+            .stringPropertyNames()
+            .stream()
+            .filter(key -> key.startsWith(SMART_TESTING_STRATEGY_PREFIX))
+            .map(String::valueOf)
+            .map(key -> key.substring(key.lastIndexOf('.') + 1))
+            .collect(Collectors.toList());
+    }
+
+    protected Properties loadDefaultMapping() {
+        final Properties properties = new Properties();
+        try (InputStream strategyMapping = getClass().getClassLoader().getResourceAsStream("strategies.properties")) {
+            if (strategyMapping == null) {
+                throw new IllegalStateException("Unable to load default strategy dependencies mapping.");
+            }
+            properties.load(strategyMapping);
+        } catch (IOException e) {
+            throw new IllegalStateException("Unable to load default strategy dependencies mapping.", e);
+        }
+        return properties;
+    }
+
+    protected Properties loadFromFile() {
+        final Properties properties = new Properties();
+        if (propertiesPath != null) {
+            try {
+                properties.load(new FileInputStream(this.propertiesPath.toFile()));
```

Comment:
I remember that we had a discussion about this functionality (autocorrection in general). I wanted to call the `StringSimilarityCalculator` on both sides Maven extension (when the dependencies are resolved) and surefire provider (when the strategies are applied) - you told me that for optimization reasons you want to call it only once. 
But now, you read the file twice - when you apply autocorrection and when you resolve dependencies. I'm afraid that reading a file is bigger load than checking similarities. WDYT?

Reply:
Yes the problem is that we need to or instantiate twice `StringSimilarityCalculator ` or we create a singleton for it.

Reply:
Looking at your changes of  `StringSimilarityCalculator` https://github.com/arquillian/smart-testing/pull/213/files#diff-61b559b9fdc0af7def1988ba0482defa there is no (and wasn't) Maven `Dependency` class used. You just take the sample (set by user) and list of targets (available strategies) and you are looking for the similarity. This is exactly what we need on both sides - Maven extension when the dependencies are resolved and surefire provider (`TestStrategyApplier`) when we apply available strategies.